vignettes/queries_dataset.Rmd

```diff
@@ -0,0 +1,178 @@
+---
+title: "Queries dataset documentation"
+output:
+  rmarkdown::html_vignette:
+  toc: false
+vignette: >
+  %\VignetteIndexEntry{Queries dataset documentation}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+  ```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+# Introduction
+
+To support the analysis of the safety analysis, it is quite common to define specific grouping of events. One of the most common way is to group events or medications by a specific medical concept such as a Standard MedDRA Queries or WHO-Drug Standardized Drug Grouping . 
+
+
+To help with the derivation of these variables, the admiral function `derive_vars_query()` can be used.
+This function takes as input the dataset (`dataset`) where the grouping must occurs (e.g `ADAE`) and
+a dataset containing the required information to perform the derivation of the grouping variables 
+(`dataset_queries`).
+
+Because the definition, collection and storage of the medical concepts is company specific, the dataset passed to the `dataset_queries` argument of the `derive_vars_query()` function should be created outside of admiral in a specific format so that it can be passed in the function.
+
+This vignette describes the expected structure and content of the dataset passed to the
+`dataset_queries` argument in the `derive_vars_query()` function. 
+
+# Structure of the Query Dataset.
+
+## Variables
+
+Variable | Scope |  Type | Example Value
+------- | ----- | ------ | ----- | ----- | -----
+**VAR_PREFIX** | The prefix used to define the grouping variables |  Character |  "SMQ01" 
+**QUERY_NAME** | The value provided to the grouping variables name|   Character | "Immune-Mediated Guillain-Barre Syndrome"
+**TERM_LEVEL** | The variable used to define the grouping. Used in conjunction with TERM_NAME |  Character | "AEDECOD"
+**TERM_NAME** | A term used to define the grouping. Used in conjunction with TERM_LEVEL |  Character | "GUILLAIN-BARRE SYNDROME"
+**TERM_ID** | A code used to define the grouping. Used in conjunction with TERM_LEVEL |  Integer | 10018767
+QUERY_ID | Id. number of the query. This could be a SMQ identifier |  Integer | 20000131
+QUERY_SCOPE | For SMQs, scope (Broad/Narrow) of the query |  Character | BROAD, NARROW, NA
+QUERY_SCOPE_NUM | For SMQs, scope (Broad/Narrow) of the query |  Integer | 1, 2, NA
+
```

Comment:
Should we clarify 1 corresponds to BROAD and 2 corresponds to NARROW, or is it user choice? If user choice, should we make a note `QUERY_SCOPE` and `QUERY_SCOPE_NUM` should match?

Reply:
That s a very good point. Currently the function only checks the values are 1,2 or NA, but there is no check that a BROAD Scope should have a `SCOPE_NUM` = 1 and a NARROW scope should be have a `SCOPE_NUM` = 2. @bundfussr - should we implement this is the `assert_valid_queries()`  function?